### PR TITLE
feat(Reports): RHINENG-13445 - Enable async filtering, sorting & pagination

### PR DESCRIPTION
**Pull request by bastilian**
This PR extends the Reports page to not only use the REST API, but also use the table state provided by the `AsyncTableToolsTable`.

This can be reviewed and tested, but some changes from https://github.com/RedHatInsights/compliance-frontend/pull/2226 that allow passing params have already been included here, excluding the tests, so the tests for `useQuery` and `useComplianceQuery` will fail till #2226 is merged and this is rebased. 

~Sorting and pagination (with more than 10 reports) works as expected, however filters are not yet properly updating, most likely due to the issue with `compileState` mention in the tests[0].~ Fixed via https://github.com/RedHatInsights/compliance-frontend/pull/2243

[0] https://github.com/RedHatInsights/compliance-frontend/blob/master/src/Frameworks/AsyncTableTools/hooks/useTableState/helpers/compileState.test.js#L38

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: src/PresentationalComponents/ReportsTable/Filters.js [hunk 1]
```diff
@@ -50,13 +50,14 @@ export const policyComplianceFilter = [
...
     type: conditionalFilterType.checkbox,
     label: 'Systems meeting compliance',
-    filterSerialiser: (_, values) =>
-      values
-        .map((value) => {
-          const scoreRange = value.split('-');
-          return `compliance_score >= ${scoreRange[0]} and compliance_score <= ${scoreRange[1]}`;
-        })
-        .join(' OR '),
+    // TODO compliance_threshold needs to be added as a search attribute in the backend
```
Comment (mkholjuraev) [inline src/PresentationalComponents/ReportsTable/Filters.js:53]: Is the backend team aware of this issue?

**Review by mkholjuraev — COMMENTED** [review]
I observed a few issues with rest and async tables are enabled: 

1. RESOLVED => I do not see the pagination dropdown. Is this intentional 
2. After all the table entities are filtered out by name (empty table), I am not able to get back to original table state with data using 'Reset filters' or clearing the filter chips.
3. Policy type filter is lost in the filters dropdown. However, I see it being passed only through GraphQL component. I think REST component misses it.
4. filtering by OS, results in page refresh on the first try. Successive applications, results in page crash.

**Comment by adonispuente** [comment]
> I observed a few issues with rest and async tables are enabled:
> 
> 1. I do not see the pagination dropdown. Is this intentional
> 2. After all the table entities are filtered out by name (empty table), I am not able to get back to original table state with data using 'Reset filters' or clearing the filter chips.
> 3. Policy type filter is lost in the filters dropdown. However, I see it being passed only through GraphQL component. I think REST component misses it.
> 4. filtering by OS, results in page refresh on the first try. Successive applications, results in page crash.

Some more findings:
5. FIltering by policy type does seem to work, but the filter chip doesnt appear, and 'clear results' isnt possible to select. 
6. System Meeting compliance filtering doesnt work.
7. When sorting by OS system, it looks like the backend returns the same order everytime, though we're send asc/desc appropriatley 
8. its hard to tell but im not 100% sure 'Clear Filters' is working properly right now

**Review by LightOfHeaven1994 — COMMENTED** [review]
I found out that with APIv2 we are still counting Compliant percentage on FE side, now if FF is enabled, we need to grab this parameter from backend response. We need to make this change [here](https://github.com/RedHatInsights/compliance-frontend/blob/master/src/SmartComponents/ReportDetails/Components/hooks/useDonutChart.js#L140) , if apiv2 enabled - grab parameter from backend, if not, count it as GQL doesn't provide us this value

**Comment by LightOfHeaven1994** [comment]
@bastilian and one more thing I forgot to mention - can we show table loading while we are waiting on backend response? right now if we do sorting, filtering table is showing old results and after some time it just updates, having loading skeleton would be nice

File: src/PresentationalComponents/ReportsTable/Columns.js [hunk 2]
```diff
@@ -35,6 +33,7 @@ export const OperatingSystem = {
 export const CompliantSystems = {
   title: 'Systems meeting compliance',
   transforms: [fitContent],
+  sortable: 'compliance_threshold',
```
[the remark this answers is not in the document]
Comment (bastilian) [reply src/PresentationalComponents/ReportsTable/Columns.js]: Thank you! Changed!

**Comment by LightOfHeaven1994** [comment]
@bastilian after last changes I see NaN is set for offset and page ends up with error
image

**Comment by bastilian** [comment]
@LightOfHeaven1994 Thank you for testing! We'll revisit the loading state later. As long as it loads data and it's correct we should be good for now. The issue with the sorting should be fixed now. 

@mkholjuraev Thank you as well for reviewing and testing! I think the only question is open is regarding policy type filter, and I think there is a Jira somewhere, where David mentioned that this would not be feasible to implement in the backend. If the backend implements it we can enable it also for the REST API. 

The table should now overall "work" there may still be some things we'll need to refine, but that will happen in working on other things or later in the polishing phase.

**Review by LightOfHeaven1994 — APPROVED** [review]
lgtm from my side, minor things will be added/fixed later. I will make notes about it

**Review by mkholjuraev — APPROVED** [review]
apart from minor issues like loading, weird columns state after column management changes, there is an issue with filtering that I observed once. The table was showing 2 reports when there we no active filters. However, as I am not able to find reproducing steps not considering as blocker.

**Comment by mkholjuraev** [comment]
@LightOfHeaven1994 filtering issue may need sooner treatment. Wanted to make sure we file a ticket for it.

**Comment by LightOfHeaven1994** [comment]
> .

I will create tickets for all things that left here, already put notes on my doc and will create tickets a bit later today or tomorrow

**Pull request merged**